### PR TITLE
Add decentralized post pinning mechanism to backend.

### PR DESCRIPTION
**Pull request by bluepartyhat**

### CODE REVIEW

File: routes/post.go
Comment (lazynina) [inline routes/post.go:83]:
```suggestion
	// True if this post hash hex is in pinned to the global feed.
```

File: routes/post.go
Comment (lazynina) [inline routes/post.go:242]: fix spacing -- don't use tabs.

File: routes/post.go
Comment (lazynina) [inline routes/post.go:996]:
```suggestion
	// Fetch the commentEntries for the post.
```

File: routes/post.go
Comment (lazynina) [inline routes/post.go:1307]: what's the significance of MaxPinnedPosts?

**Review by lazynina — CHANGES_REQUESTED** [review]

File: routes/post.go
Comment (lazynina) [reply routes/post.go:1307]: I think this can be rewritten to support returning all pinned posts? I tested pinning 20+ posts and only the top 5 appear as pinned.

File: routes/post.go
Comment (lazynina) [inline routes/post.go:1376]: Let's add a comment explaining MaxPinnedPosts

**Review by lazynina — APPROVED** [review]

**Comment by lazynina** [comment]
Need to run CI after core gets merged in to make sure tests pass.

**Comment by chafreaky** [comment]
Bump on this. Barring the conflict, is there anything holding up this merge? It's a great feature!

**Pull request closed without merging**
